### PR TITLE
Facebook disconnect support

### PR DESCRIPTION
**Pull request by drsnyder**
This PR adds support for disconnecting/unlinking a facebook user from a Wikia user. 

https://wikia-inc.atlassian.net/browse/SERVICES-623

TODO:
 * [x] add controller unit tests for the new functionality
 * [ ] add integration tests
 * [ ] update the swagger documentation

### CODE REVIEW

File: service/facebook.go [hunk 1]
```diff
@@ -53,6 +54,20 @@ func (facebookService *FacebookServiceValue) ConnectUsers(userId, appId int64, a
 	return saved, errors
 }
 
+func (facebookService *FacebookServiceValue) DisconnectUsers(userId, appId int64) (bool, *errorlist.ErrorList) {
+	_, errors := facebookService.userService.LookupUserById(userId)
```
Comment (drsnyder) [inline service/facebook.go]: I don't think we need to validate the facebook access token here. It should be sufficient if the wikia user is authenticated.

File: controllers/users_controller.go [hunk 2]
```diff
@@ -183,6 +185,29 @@ func (controller *UsersController) connectFacebookUser(w http.ResponseWriter, r
...
 
+func (controller *UsersController) disconnectFacebookUser(w http.ResponseWriter, r *http.Request, params map[string]string) {
+	validationErrors := errorlist.NewErrorList()
+	userId := parseInt64(params["user_id"])
+	facebookAppId := parseInt64(params["facebook_app_id"])
+	requestingUserId := getAuthenticatedUserId(r)
+
+	if !authorizeAction(requestingUserId, userId) {
+		validationErrors.Add(service.FacebookMappingAuthorizationError)
+		utils.UnauthorizedErrorList("You are not authorized to connect this user", validationErrors).Write(w)
```
Comment (nmonterroso) [inline controllers/users_controller.go]: should this be "disconnect"?

File: service/facebook.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ var (
 
 type FacebookService interface {
 	ConnectUsers(userId, appId int64, accessToken string) (bool, *errorlist.ErrorList)
+	DisconnectUsers(userId, appId int64) (bool, *errorlist.ErrorList)
```
Comment (nmonterroso) [inline service/facebook.go]: The plural here is strange, since it's only one user. I guess it's consistent with `ConnectUsers`, but still a bit weird

**Comment by nmonterroso** [comment]
minor comments otherwise LGTM

File: controllers/users_controller.go (hunk 2 above)
Comment (drsnyder) [inline controllers/users_controller.go]: Yeah, fixed in 73582bc

File: service/facebook.go (hunk 3 above)
Comment (harnash) [inline service/facebook.go]: Fixed in 6cf2398.

**Comment by nmonterroso** [comment]
Are all the TODOs done?

**Comment by nmonterroso** [comment]
also remember to run `./generateResources.sh`

**Comment by harnash** [comment]
Resources regenerated.

**Comment by harnash** [comment]
Problem is with integration tests since we don't actually have any for Facebook integration...

**Comment by drsnyder** [comment]
> Problem is with integration tests since we don't actually have any for Facebook integration...

I would be ok ticketing this to follow up on.

**Comment by drsnyder** [comment]
The only failing test is this one:
```
--- FAIL: TestUpdateUserPreference (0.17s)
	assertions.go:150:
	Location:	user_preference_e2e_test.go:30
	Error:		Should be true
	Messages:	failed to update
	assertions.go:150:
	Location:	user_preference_e2e_test.go:31
	Error:		Expected nil, but got: &errorlist.KeyedError{Key:"server", Message:"user_preference_service_error"}	
	Messages:	An Error occur
```
It's also failing in my environment. I'll investigate.

**Comment by drsnyder** [comment]
Ok, fixing the tag doesn't fix it because the qa boxes are in the production environment.

```
damon@qa-vm20:~$ echo $WIKIA_ENVIRONMENT 
prod
```

**Comment by drsnyder** [comment]
Actually those boxes aren't running the consul agent. Working with ops to configure them with consul in the dev datacenter.

**Comment by drsnyder** [comment]
OPS is adding the consul agent to the qa boxes [here](https://github.com/Wikia/chef-repo/pull/6295) in the dev datacenter. Once that's done I'll try re-running the tests.

**Pull request merged**